### PR TITLE
Operator upgrade guide

### LINKED ISSUE #863: Scylla cluster fails to be rolled out after operator upgrade
**Describe the bug**
If we upgrade operator from the v1.5.0 version to the v1.6.0-rc.0 one. Then existing Scylla clusters fail to be rolled out with the following error:

```
I1129 09:31:21.061931       1 operator/sidecar.go:218] "Waiting for single service informer caches to sync"
I1129 09:31:21.162317       1 operator/sidecar.go:235] "Waiting for Service" Service="scylla/sct-cluster-us-east1-b-us-east1-2"
I1129 09:31:21.166368       1 operator/sidecar.go:269] "Waiting for Pod To have scylla ContainerID set" Pod="scylla/sct-cluster-us-east1-b-us-east1-2"
E1129 09:31:21.170105       1 cache/reflector.go:138] k8s.io/client-go@v0.22.3/tools/cache/reflector.go:167: Failed to watch *v1.Pod: unknown (get pods)
I1129 09:31:21.170175       1 operator/sidecar.go:323] "Waiting for NodeConfig's data ConfigMap " Selector="scylla-operator.scylladb.com/config-map-type=NodeConfigData,scylla-operator.scylladb.com/owner-uid=ac58be37-ed23-454f-8267-f34ae09ef5fe"
I1129 09:38:13.055910       1 signals/signal.go:26] Received shutdown signal: terminated; shutting down...
Error: can't wait for optimization: timed out waiting for the condition
```

**To Reproduce**
1. Deploy operator v1.5.0 (latest stable)
2. Deploy a Scylla cluster
3. Upgrade operator to v1.6.0-rc.0

**Expected behavior**
Scylla cluster must successfully get rolled out after operator upgrade.

**Logs**
[EKS logs](https://cloudius-jenkins-test.s3.amazonaws.com/8cd4b20d-dd42-48eb-99c3-46da6f12817e/20211129_093839/kubernetes-8cd4b20d.tar.gz)
[GKE logs](https://cloudius-jenkins-test.s3.amazonaws.com/1a3969dc-fded-46d5-8eaa-34dcfe6f195f/20211129_094625/kubernetes-1a3969dc.tar.gz)

**Environment:**
 - Platform: EKS, GKE
 - Kubernetes version: v1.20 (may be any version)
 - Scylla version: 4.4.5
 - Scylla-operator version: v1.6.0-rc.0

**Additional context**

### PR DESCRIPTION
**Pull request by zimnx**
**Description of your changes:**

Helm is not able to manage CRDs - it only creates them on first intall. To make sure all available CRDs are installed and up-to-date we have to add additional step to upgrade guide.

**Which issue is resolved by this Pull Request:**
Resolves #863

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -418,10 +418,12 @@ update-deploy:
 	$(call generate-operator-manifests,helm/deploy/operator.yaml,deploy/operator,$(tmp_dir))
 	$(call generate-manager-manifests-prod,helm/deploy/manager_prod.yaml,deploy/manager/prod,$(tmp_dir))
 	$(call generate-manager-manifests-dev,deploy/manager/dev)
+	$(call concat-manifests,$(sort $(wildcard deploy/operator/00_scylla.scylladb.com_*.yaml)),deploy/operator.crds.yaml)
```
Comment (tnozicka) [inline Makefile:421]: deploy folder should contain 1 manifest per file

File: docs/source/upgrade.md [hunk 2]
```diff
@@ -2,6 +2,24 @@
 
 This pages describes Scylla Operator upgrade procedures.
 
+## Upgrade via Helm
```
Comment (tnozicka) [inline docs/source/upgrade.md:5]: Add few sentences explaining that helm can't handle the CRD update, possibly links

File: docs/source/upgrade.md [hunk 3]
```diff
@@ -2,6 +2,24 @@
...
+Replace `<release_name>` with the name of your Helm release for Scylla Operator and 
+replace `<version>` with the version number you want to install:
+1. Make sure Helm chart repository is up-to-date:
+   ```
+   helm repo add scylla-operator https://storage.googleapis.com/scylla-operator-charts/stable
+   helm repo update
+   ```
+2. Update CRD resources
+   ```
+   tmpdir=$( mktemp -d ) && helm pull scylla-operator/scylla-operator --version <version> --untar --untardir "${tmpdir}" && find "${tmpdir}"/scylla-operator/crds/ -name '*.yaml' -printf '-f=%p ' | xargs kubectl apply
```
Comment (tnozicka) [inline docs/source/upgrade.md:15]: `--server-side` (is important for more actors, maybe a note it's fine without it for old versions that don't have it)

File: docs/source/upgrade.md (hunk 2 above)
Comment (zimnx) [reply docs/source/upgrade.md:7]: done

File: docs/source/upgrade.md (hunk 3 above)
Comment (zimnx) [reply docs/source/upgrade.md:20]: good point, done

File: docs/source/upgrade.md [hunk 4]
```diff
@@ -1,6 +1,47 @@
 # Upgrade of Scylla Operator
 
-This pages describes Scylla Operator upgrade procedures.
+This pages describes Scylla Operator upgrade procedures.
```
Comment (tnozicka) [inline docs/source/upgrade.md:3]:
```suggestion
This page describes Scylla Operator upgrade procedures.  
```

File: docs/source/upgrade.md [hunk 5]
```diff
@@ -1,6 +1,47 @@
 # Upgrade of Scylla Operator
 
-This pages describes Scylla Operator upgrade procedures.
+This pages describes Scylla Operator upgrade procedures.  
+There are two generic update procedures - via Helm and via kubectl. Before upgrading check if this page contains steps for version you want to update.
+If not, follow the generic guide.
```
Comment (tnozicka) [inline docs/source/upgrade.md:5]: (accounted for with "additional steps")

File: docs/source/upgrade.md [hunk 6]
```diff
@@ -1,6 +1,47 @@
...
+
+1. Checkout source code of version you want to use:
+   ```
+   git checkout <version>
+   ```
+2. Manifests use rolling minor version tag, you may want to pin it to specific version:
+   ```
+   find deploy/operator -name "*.yaml" | xargs sed --follow-symlinks -i -E "s^docker.io/scylladb/scylla-operator:[0-9]+\.[0-9]+^docker.io/scylladb/scylla-operator:<version>^g"
+   ```
+3. Update Scylla Operator (newer versions of kubectl may additionally add `--server-side` parameter)
```
Comment (tnozicka) [inline docs/source/upgrade.md:39]: same here

File: docs/source/upgrade.md (hunk 4 above)
Comment (zimnx) [reply docs/source/upgrade.md]: done

File: docs/source/upgrade.md [hunk 7]
```diff
@@ -1,6 +1,47 @@
 # Upgrade of Scylla Operator
 
-This pages describes Scylla Operator upgrade procedures.
+This pages describes Scylla Operator upgrade procedures.  
+There are two generic update procedures - via Helm and via kubectl. Before upgrading check if this page contains steps for version you want to update.
```
[the remark this answers is not in the document]
Comment (zimnx) [reply docs/source/upgrade.md]: done

File: docs/source/upgrade.md (hunk 5 above)
Comment (zimnx) [reply docs/source/upgrade.md]: done

File: docs/source/upgrade.md [hunk 8]
```diff
@@ -1,6 +1,47 @@
...
+Helm doesn't support managing CustomResourceDefinition resources ([#5871](https://github.com/helm/helm/issues/5871), [#7735](https://github.com/helm/helm/issues/7735))   
+These are only created on first install and never updated. In order to update them, users have to do it manually.
+
+Replace `<release_name>` with the name of your Helm release for Scylla Operator and replace `<version>` with the version number you want to install:
+1. Make sure Helm chart repository is up-to-date:
+   ```
+   helm repo add scylla-operator https://storage.googleapis.com/scylla-operator-charts/stable
+   helm repo update
+   ```
+2. Update CRD resources (newer versions of kubectl may additionally add `--server-side` parameter)
```
[the remark this answers is not in the document]
Comment (zimnx) [reply docs/source/upgrade.md]: done

File: docs/source/upgrade.md (hunk 3 above)
Comment (zimnx) [reply docs/source/upgrade.md]: done

File: docs/source/upgrade.md (hunk 6 above)
Comment (zimnx) [reply docs/source/upgrade.md]: done

File: docs/source/upgrade.md [hunk 9]
```diff
@@ -1,6 +1,50 @@
...
+
+1. Checkout source code of version you want to use:
+   ```
+   git checkout <version>
+   ```
+2. Manifests use rolling minor version tag, you may want to pin it to specific version:
+   ```
+   find deploy/operator -name "*.yaml" | xargs sed --follow-symlinks -i -E "s^docker.io/scylladb/scylla-operator:[0-9]+\.[0-9]+^docker.io/scylladb/scylla-operator:<version>^g"
+   ```
+3. Update Scylla Operator We recommend using `--server-side` flag for `kubectl apply`, if your version supports it.
```
Comment (tnozicka) [inline docs/source/upgrade.md:42]:
```suggestion
3. Update Scylla Operator. We recommend using `--server-side` flag for `kubectl apply`, if your version supports it.
```

**Review by tnozicka — COMMENTED** [review]
last nit

**Review by tnozicka — APPROVED** [review]

**Comment by tnozicka** [comment]
/lgtm
thanks

**Comment by zimnx** [comment]
clean rebase

**Review by tnozicka — APPROVED** [review]

**Pull request merged**